core/shared/src/main/scala/zio/internal/FiberContext.scala

```diff
@@ -522,6 +574,9 @@ private[zio] final class FiberContext[E, A](
 
                   curZio = nextInstr(result)
 
+                case ZIO.Tags.RaceWith =>
+                  val zio = curZio.asInstanceOf[ZIO.RaceWith[Any, Any, Any, Any, Any, Any, Any]]
```

Comment:
You probably don't need this cast because `raceWithImpl` is polymorphic.

Reply:
`raceWithImpl` needs a `RaceWith` instance. `curZio` would be `IO[E, Any]`